flink-annotations/src/main/java/org/apache/flink/annotation/Public.java

```diff
@@ -35,4 +37,5 @@
 @Documented
 @Target(ElementType.TYPE)
 @Public
+@Retention(RetentionPolicy.RUNTIME)
```

Comment:
Do we actually need?
We don't have it for others IIUC

Reply:
We do - its needed for the annotation alignment tests so that the tests can see what annotations classes/functions have at runtime.